### PR TITLE
Minor updates related to skd crash handling

### PR DESCRIPTION
**Pull request by jeffli-xilinx**
#### Problem solved by the commit
Missed last commit for PR comments

### CODE REVIEW

**Review by sonals — CHANGES_REQUESTED** [review]
1. Please update signatures of all relevant functions to use `enum kds_status` instead of `int status`
2. Remove all the `status` range checks like `if ((status >=  0) && (status < KDS_STAT_MAX))`

File: src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c [hunk 1]
```diff
@@ -404,37 +404,22 @@ int zcu_xgq_unassign_cu(struct platform_device *pdev, u32 cu_idx, u32 cu_domain)
...
-		init_resp_abort(&r, cmd->cid, ret, status);
-	else
-		init_resp(&r, cmd->cid, ret);
+	if (unlikely(!(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq)))) {
+		rptr = &r;
+		r.hdr.cid = cmd->cid;
+		r.hdr.cstate = XGQ_CMD_STATE_COMPLETED;
+		r.result = status;
+		r.rcode = ret;
+		if (unlikely(status == KDS_SKCRASHED))
```
Comment (sonals) [inline src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c:420]: `r.hdr.cstate = (status == KDS_SKCRASHED) ? XGQ_CMD_STATE_ABORTED : XGQ_CMD_STATE_COMPLETED`

and remove the previous r.hdr.cstate assignment.

File: src/runtime_src/core/edge/skd/sk_daemon.cpp [hunk 2]
```diff
@@ -69,7 +69,8 @@ static void sigLog(const int sig)
 #endif
 	skd_inst->report_crash();
 	xrt_core::message::send(severity_level::error, "SKD", "SKD Reported crash!");
-	exit(128 + sig); // Convert to Linux exit code
+	int exit_code = 128 + sig; // Linux exit code is always 128 + signal number
```
Comment (sonals) [inline src/runtime_src/core/edge/skd/sk_daemon.cpp:72]: This does not add any value. Please remove.

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c [hunk 3]
```diff
@@ -454,7 +453,7 @@ static void notify_execbuf(struct kds_command *xcmd, int status)
 		client_stat_inc(client, c_cnt[xcmd->cu_idx]);
 
 	if (xcmd->inkern_cb) {
-		int error = (status == ERT_CMD_STATE_COMPLETED)?0:-EFAULT;
+		int error = (status == KDS_COMPLETED)?0:-EFAULT;
```
Comment (sonals) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c:456]: Add white space before/after operators:
` (status == KDS_COMPLETED) ? 0 : -EFAULT;`

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c [hunk 4]
```diff
@@ -502,7 +501,7 @@ static void notify_execbuf_xgq(struct kds_command *xcmd, int status)
...
```
Comment (sonals) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c:504]: Add white space as above...

File: src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c [hunk 5]
```diff
@@ -404,42 +404,25 @@ int zcu_xgq_unassign_cu(struct platform_device *pdev, u32 cu_idx, u32 cu_domain)
...
-		rptr = NULL;
-	else if (unlikely(status == KDS_SKCRASHED))
-		init_resp_abort(&r, cmd->cid, ret, status);
-	else
-		init_resp(&r, cmd->cid, ret);
+	if (unlikely(!(ret == 0 && ZCU_XGQ_FAST_PATH(zcu_xgq)))) {
+		rptr = &r;
+		r.hdr.cid = cmd->cid;
+		r.result = status;
+		r.rcode = ret;
```
Comment (sonals) [inline src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c:418]: rcode is unsigned int and we are storing signed int.
Should **rcode** be **s32**?

**Review by sonals — APPROVED** [review]

File: src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c [hunk 6]
```diff
@@ -404,42 +404,25 @@ int zcu_xgq_unassign_cu(struct platform_device *pdev, u32 cu_idx, u32 cu_domain)
...
-	resp->rcode = rcode;
-}
-
 static inline void
-zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, int ret, int status)
+zcu_xgq_cmd_complete(struct platform_device *pdev, struct xgq_cmd_sq_hdr *cmd, uint ret, enum kds_status status)
 {
-	struct xgq_com_queue_entry r;
-	struct xgq_com_queue_entry *rptr = &r;
+	struct xgq_com_queue_entry r = {0};
```
Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c:410]: Zero not needed
```suggestion
	struct xgq_com_queue_entry r = {};
```

File: src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c [hunk 7]
```diff
@@ -459,16 +442,16 @@ static inline void
 zcu_xgq_cmd_start_cuidx(struct zocl_cu_xgq *zcu_xgq, struct xgq_cmd_sq_hdr *cmd)
 {
 	struct kds_command *xcmd;
-	int mask_idx;
-	int bit_idx;
+	int mask_idx = 0;
```
Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c:445]: Thank you for initializing the variables.

File: src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c [hunk 8]
```diff
@@ -459,16 +442,16 @@ static inline void
...
+	int bit_idx = 0;
 
 #if 0
```
Comment (rozumx) [inline src/runtime_src/core/edge/drm/zocl/zocl_cu_xgq.c:448]: Is this #if statement still needed.  If not, then maybe this code should be removed.

File: src/runtime_src/core/pcie/driver/linux/xocl/subdev/cu_xgq.c [hunk 9]
```diff
@@ -59,12 +59,7 @@ static void cu_xgq_check(void *core, struct xcu_status *status, bool force)
...
 
 	status->num_ready = 1;
-
-	do {
-		ret = xocl_xgq_check_response(cu_xgq->xgq, cu_xgq->xgq_client_id, &cu_status);
-	}
-	while (!ret);
-
+	while (!xocl_xgq_check_response(cu_xgq->xgq, cu_xgq->xgq_client_id, &cu_status));
 	status->new_status = (cu_status == KDS_SKCRASHED) ? CU_AP_CRASHED : CU_AP_IDLE;
```
Comment (rozumx) [inline src/runtime_src/core/pcie/driver/linux/xocl/subdev/cu_xgq.c:63]: Add a new line before this statement to help prevent future developers from thinking that this line is associated with the above while loop.
```suggestion
	
	status->new_status = (cu_status == KDS_SKCRASHED) ? CU_AP_CRASHED : CU_AP_IDLE;
```

**Review by rozumx — APPROVED** [review]
Approved with some minor future pull request recommentations.

**Pull request merged**